don't display continue mapping button

Hagellach37 opened the pull request:
Do not display the "continue mapping" button at the end of a mapping session, when the user has already mapped all groups of the 15 groups that are available. In that situation the user can only use the "Complete Session" button and will be navigated back to the projectView screen.

![image](https://user-images.githubusercontent.com/7045979/126489874-e0a6bad0-1dcb-4184-b94c-6f04d9ed7bf7.png)

Hagellach37 commented:
I do not understand why the Android build failed.

laurentS commented on src/shared/views/ChangeDetection/TaskList.js:
```diff
@@ -40,6 +40,7 @@ type Props = {
     closeTilePopup: () => void,
     openTilePopup: () => void,
     zoomLevel: number,
+    continueMapping: boolean,
```
Could you maybe rename this to `canContinueMapping`? I think it makes it clearer what the variable means, no?

Hagellach37 replied:
good idea. will rename this in all places where it occurs

laurentS commented on src/shared/common/firebaseFunctions.js:
```diff
@@ -126,6 +131,10 @@ export const mapStateToPropsForGroups =
             const groupsToPickFrom = groupsAvailable.filter(
                 g => !groupsMapped.includes(g),
             );
+            if (groupsToPickFrom.length === 1) {
```
Is there any possibility that the length of the list drops from 2 to 0 without ever being 1? I don't think so, but just double checking :)

Hagellach37 replied:
good catch. I will change to <= 1

laurentS commented on src/shared/common/firebaseFunctions.js:
```diff
@@ -115,6 +115,11 @@ export const mapStateToPropsForGroups =
+        // Here we define a boolean value 'continueMapping' which will tell us,
+        // if there are more groups available for mapping for this user.
+        let continueMapping;
+        continueMapping = true;
```
Any reason not to write `let continueMapping = true`?

Hagellach37 replied:
no particular reason. so I will adjust as you recommended

laurentS reviewed (changes requested):
Looks cool, I like the idea! Just a couple of minor comments around naming things to avoid confusion.

Hagellach37 commented:
Hey @laurentS ,

I've adjusted this based on your comments. I also added this for the building footprint project type. When you want to test this locally, it is handy if you do not get 15 groups but only 3 or so. You can adjust this in `firebaseFunctions.js`

Hagellach37 commented:
closing this in favor of #571
